### PR TITLE
feat: upgrade to 1.64

### LINKED ISSUE #4822: [Chore]: update vscode to 1.64
https://code.visualstudio.com/updates/v1_64

### PR DESCRIPTION
**Pull request by jsjoeio**
# Summary
This PR upgrades VS Code to 1.64.

## Testing Plan

To test this locally, you need to:
1. clone `coder/code-server` locally
2. make sure `yarn link` is set up correctly
3. run `yarn watch`

## Video 

https://user-images.githubusercontent.com/3806031/155605927-fb75678c-e1b2-4aef-a009-5953e21369db.mov

## How to Review this PR
I committed the merge conflicts and fixed them one-by-one. Look at the Commits tab starting with [this commit](https://github.com/coder/vscode/pull/45/commits/84ad4c13015a5d3e0b1b5f9c1a650b7c54c75d10).

## Files with Merge Conflicts
  - [x] package.json 
  - [x] yarn.lock
  - [x] product.json 
  - [x] scripts/code-server.sh 
  - [x] src/vs/base/common/network.ts 
  - [x] src/vs/code/browser/workbench/workbench-dev.html 
  - [x] src/vs/code/browser/workbench/workbench.html 
  - [x] src/vs/platform/remote/browser/browserSocketFactory.ts 
  - [x] src/vs/platform/remote/common/remoteAgentConnection.ts 
  - [x] src/vs/server/node/serverEnvironmentService.ts 
  - [x] src/vs/workbench/browser/web.main.ts 
  - [x] src/vs/workbench/browser/web.api.ts 
  - [x] src/vs/server/node/server.main.ts 
  - [x] src/vs/workbench/contrib/extensions/browser/extensionsViewlet.ts 
  - [x] src/vs/server/node/webClientServer.ts 
  - [x] src/vs/code/browser/workbench/workbench.ts 
  - [x] src/vs/server/node/remoteExtensionHostAgentServer.ts 

## Next Steps

- [x] Look at todos in code and remove one by one
- [x] Test main code-server features 
- [x] Clean up
- [x] Check that URL uses short version with folder path
- [x] double check that last-opened workspace still works (try cli flag)
- [x] open a webview (like an image file)
- [x] set loglevel `--trace` and open browser console
- [x] Talk to Asher about checking patches in GitHub UI to make sure we have them (?)
- [x] Fix folder path in URL: https://github.com/coder/code-server/pull/4928
- [x] Fix opening Folder or Workspace from CLI: https://github.com/coder/code-server/pull/4932
- [x] remove all TODOS

Fixes https://github.com/coder/code-server/issues/4822

### CODE REVIEW

**Comment by jsjoeio** [comment]
>  src/vs/code/browser/workbench/workbench.ts

Should be the next thing to work on.

**Comment by jsjoeio** [comment]
### Folder Path in URL

Here is what happens when you open code-server. 

`http://localhost:3000/?folder=%2FUsers%2Fjp%2Fdev%2Fcoder` - I think we need to add our patch back then.

**Comment by jsjoeio** [comment]
## Opening Folder or Workspace from CLI 🐛

Testing locally with `yarn watch --port 3000 ./docs` and it did not work. It seems to only open your last opened workspace.

**Review by code-asher — APPROVED** [review]
Nice work!  Everything looks good to me.

File: src/vs/workbench/browser/client.ts [hunk 1]
```diff
@@ -29,6 +29,7 @@ export class CodeServerClientAdditions extends Disposable {
 	static LOGOUT_COMMAND_ID = 'code-server.logout';
 	static AUTH_KEY = 'code-server.authed';
 	productConfiguration: Partial<IProductConfiguration>;
+	ctor: any;
```
[the remark this answers is not in the document]
Comment (code-asher) [reply src/vs/workbench/browser/client.ts:32]: Oh whoops I think this was removed in a later commit, please ignore

[the remark this answers is not in the document]
Comment (code-asher) [reply src/vs/workbench/browser/client.ts:32]: Or maybe not, but we can take a look later

File: scripts/code-server.sh [hunk 2]
```diff
@@ -13,17 +13,14 @@ function code() {
...
 
-<<<<<<< HEAD:resources/server/web.sh
-	# # Load remote node
-	# yarn gulp node
-
-=======
->>>>>>> upstream/release/1.64:scripts/code-server.sh
-	NODE=$(node build/lib/node.js)
-	if [ ! -e $NODE ];then
+	# NOTE@coder: Never load the remote node
```
[the remark this answers is not in the document]
Comment (jsjoeio) [reply scripts/code-server.sh:16]: Ah got it. I will revert then.

**Pull request merged**